### PR TITLE
Feat typescript frontend

### PR DESCRIPTION
**Pull request by NeilChen0812**

### CODE REVIEW

File: app/src/utils/errors.ts [hunk 1]
```diff
@@ -1,35 +1,37 @@
 /* eslint-disable max-classes-per-file */
 export class TooManyRequestsError extends Error {
-  constructor(message) {
+  constructor(message: string | undefined) {
```
Comment (lancatlin) [inline app/src/utils/errors.ts:3]: You can simply use optional parameter:
```suggestion
  constructor(message?: string) {
```

File: app/src/utils/socket.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
 import io from 'socket.io-client'
 
 class SocketWrapper {
+  socket: any
```
Comment (lancatlin) [inline app/src/utils/socket.ts:4]: Find the correct type for socket.io client

**Review by lancatlin — CHANGES_REQUESTED** [review]
Prevent using `any` as type. Define interface for the axios request and response. Ref: https://bobbyhadz.com/blog/typescript-http-request-axios

**Comment by Annie04082020** [comment]
Remember to change type "any" to specify ones

**Review by lancatlin — APPROVED** [review]

**Pull request merged**